src/DynamoCore/Core/Threading/DynamoScheduler.cs

```diff
@@ -170,9 +182,10 @@ public void ScheduleForExecution(AsyncTask asyncTask)
         }
 
         /// <summary>
-        ///     Flag determining whether or not the scheduler is operating in Test Mode.
+        /// Flag determining whether or not the scheduleed task will be
+        /// processed immediately or not. 
         /// </summary>
-        public bool IsTestMode { get; private set; }
+        public TaskProcessMode ProcessMode { get; set; }
```

Comment:
My initial thought is to specify that in `StartConfiguration`, you could see that I create a virtual method in `DynamoModelTestBase.CreateStartConfiguation()` so that unit test could provide its own configurations. 

We should also remove `DynamoModel.IsTestMode`, it doesn't make sense to keep this global flag in model.